### PR TITLE
Enable Windows support

### PR DESCRIPTION
**Pull request by simonbyrne**
I'm not quite sure, but these look like the main things needed for Windows support. Would it be possible to try it out on staging?

cc: @travis-ci/julia-maintainers

### CODE REVIEW

**Comment by BanzaiMan** [comment]
This has been deployed to https://staging.travis-ci.com.

**Comment by simonbyrne** [comment]
Ah looks like I need to enable it elsewhere. Can we try it again?

**Comment by BanzaiMan** [comment]
@simonbyrne it's deployed now

**Comment by simonbyrne** [comment]
Hmm, not working: https://staging.travis-ci.org/JuliaCI/TravisTest.jl/jobs/731547

**Comment by BanzaiMan** [comment]
You are using .org staging. It was deployed to .com staging. Sorry about the miscommunication.

**Comment by simonbyrne** [comment]
is windows only available on .com?

**Comment by BanzaiMan** [comment]
No. It's also available on .org. I've deployed this to .org staging as well now. Try restarting your builds.

**Comment by simonbyrne** [comment]
Ah, I think I forgot to make it executable. Can we try that again, thanks?

**Comment by BanzaiMan** [comment]
Deployed

**Comment by simonbyrne** [comment]
Ah, my apologies! Turns out I had left some powershell flags in there. Can we try this once more?

**Comment by BanzaiMan** [comment]
Deployed.

**Comment by simonbyrne** [comment]
Hmm, it still seems to be hanging:
https://staging.travis-ci.org/JuliaCI/TravisTest.jl/jobs/731548

@BanzaiMan Thanks for your help, I'll try to look into this more later.

**Comment by BanzaiMan** [comment]
Could very well be slow I/O. We've had a few such reports in https://travis-ci.community/c/environments/windows.

**Comment by andreasnoack** [comment]
Bump. Would be great to have Windows available.

**Comment by BanzaiMan** [comment]
This has been deployed to https://staging.travis-ci.com/. I apologize in advance that this may have to be deployed over without notice.

**Comment by simonbyrne** [comment]
I just tried it (https://staging.travis-ci.org/JuliaCI/TravisTest.jl/jobs/731548), and got:
```
0.17s$ CURL_USER_AGENT="Travis-CI $(curl --version | head -n 1)"
Operating system not supported: windows
From https://github.com/JuliaCI/TravisTest.jl
```

**Comment by BanzaiMan** [comment]
@simonbyrne That's on .org. Please use https://staging.travis-ci.com.

**Comment by simonbyrne** [comment]
I tried setting it up there, but it asks about a subscription?

**Comment by ararslan** [comment]
Bump @BanzaiMan, when you have a moment could we get some guidance for how to test this given Simon's comment above?

File: lib/travis/build/script/julia.rb [hunk 1]
```diff
@@ -57,7 +57,7 @@ def setup
             when 'windows'
               sh.cmd %Q{curl -A "$CURL_USER_AGENT" -s -L --retry 7 -o julia-installer.exe '#{julia_url}'}
               sh.cmd 'chmod +x julia-installer.exe'
-              sh.cmd %Q{cmd.exe /c 'julia-installer.exe /S /D=C:/julia'}
+              sh.cmd %Q{pwsh.exe -c 'Start-Process -FilePath julia-installer.exe -ArgumentList "/D=C:/julia" -NoNewWindow -Wait'}
```
Comment (Gnimuc) [inline lib/travis/build/script/julia.rb]: `sh.cmd %Q{pwsh.exe -c 'Start-Process -FilePath julia-installer.exe -ArgumentList "/S /D=C:/julia" -NoNewWindow -Wait'}`

It looks like both `/S` and `-NoNewWindow` are needed.

**Comment by ararslan** [comment]
Successful Windows build! https://staging.travis-ci.org/ararslan/Jackknife.jl/builds/744210

**Pull request merged**